src/mcpbr/infrastructure/manager.py

```
@@ -0,0 +1,114 @@
+            mode = getattr(config, "infrastructure_mode", "local")
+        else:
+            mode = infra_config.mode
+
+        if mode == "local":
+            return LocalProvider()
+        elif mode == "azure":
+            from .azure import AzureProvider
+
+            return AzureProvider(config)
+        else:
+            raise ValueError(f"Unknown infrastructure mode: {mode}")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)